src/pages-and-resources/discussions/app-config-form/apps/shared/discussion-topics/DiscussionTopics.jsx

```diff
@@ -0,0 +1,88 @@
+import React, { useState, useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { Add } from '@edx/paragon/icons';
+import { Button } from '@edx/paragon';
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import { FieldArray, useFormikContext } from 'formik';
+import { v4 as uuid } from 'uuid';
+
+import messages from '../messages';
+import TopicItem from './TopicItem';
+import { removeModel } from '../../../../../../generic/model-store';
+import { updateDiscussionTopicIds } from '../../../../data/slice';
+import { updatedDiscussionTopics } from '../../../../data/thunks';
+
+const DiscussionTopics = ({ intl }) => {
+  const dispatch = useDispatch();
+  const { values } = useFormikContext();
+  const [topics, setTopics] = useState(values.discussionTopics);
+
+  useEffect(() => {
+    const updatedDiscussionTopicIds = values.discussionTopics?.map(topic => topic.id);
```

Comment:
what will happen if discussionTopics are undefined?  what will pass in updatedDiscussionTopicIds?

Reply:
discussionTopics will not be undefined.